iceoryx_utils/platform/win/source/time.cpp

```diff
@@ -0,0 +1,187 @@
+        waitUntil.tv_nsec += timerid->parameter.timeParameters.it_value.tv_nsec;
+        timerid->parameter.wakeup.wait_until(ulock, timePoint_t(getNanoSeconds(waitUntil)), [timerid] {
+            return !timerid->parameter.isTimerRunning || !timerid->parameter.keepRunning;
+        });
+    }
+    else
+    {
+        timerid->parameter.wakeup.wait(
+            ulock, [timerid] { return timerid->parameter.isTimerRunning || !timerid->parameter.keepRunning; });
+    }
+
+    return timerid->parameter.isTimerRunning;
+}
+
+static void
+setTimeParameters(timer_t timerid, const itimerspec& timeParameters, const bool runOnce, const bool isTimerRunning)
+{
+    std::lock_guard<std::mutex> l(timerid->parameter.mutex);
+    clock_gettime(CLOCK_REALTIME, &timerid->parameter.startTime);
+    timerid->parameter.timeParameters = timeParameters;
+    timerid->parameter.runOnce = runOnce;
+    timerid->parameter.wasCallbackCalled = false;
+    timerid->parameter.isTimerRunning = isTimerRunning;
+}
+
+int timer_create(clockid_t, struct sigevent* sevp, timer_t* timerid)
+{
+    timer_t timer = new appleTimer_t();
+    timer->callback = sevp->sigev_notify_function;
+    timer->callbackParameter = sevp->sigev_value;
+
+    timer->thread = std::thread([timer] {
+        while ([&]() -> bool {
+            std::lock_guard<std::mutex> lock(timer->parameter.mutex);
+            return timer->parameter.keepRunning;
+        }())
+        {
+            if (waitForExecution(timer))
+            {
+                timer->parameter.mutex.lock();
+                bool doCallCallback =
+                    !timer->parameter.runOnce || (timer->parameter.runOnce && !timer->parameter.wasCallbackCalled);
+                timer->parameter.mutex.unlock();
+                if (doCallCallback)
+                {
+                    std::thread t(timer->callback, timer->callbackParameter);
+                    t.detach();
+
+                    std::lock_guard<std::mutex> l(timer->parameter.mutex);
+                    timer->parameter.wasCallbackCalled = true;
+                }
+            }
+        }
+    });
+
+    *timerid = timer;
+    return 0;
+}
+
+int timer_delete(timer_t timerid)
+{
+    stopTimerThread(timerid);
+    delete timerid;
+    return 0;
+}
+
+int timer_settime(timer_t timerid, int, const struct itimerspec* new_value, struct itimerspec*)
+{
+    // disarm timer
+    if (new_value->it_value.tv_sec == 0 && new_value->it_value.tv_nsec == 0)
+    {
+        setTimeParameters(timerid, *new_value, false, false);
+    }
+    // run once
+    else if (new_value->it_interval.tv_sec == 0 && new_value->it_interval.tv_nsec == 0)
+    {
+        setTimeParameters(timerid, *new_value, true, true);
+    }
+    // run periodically
+    else
+    {
+        setTimeParameters(timerid, *new_value, false, true);
+    }
+    timerid->parameter.wakeup.notify_one();
+
+    return 0;
+}
+
+int timer_gettime(timer_t timerid, struct itimerspec* curr_value)
```

Comment:
The mismatch in variable naming convention is making me cringe. Is this mimicing a system API ? We have camelCase, snake_case and lowercase mixed in the same function !

Reply:
Yes you are right but we have to reimplement the posix functions like `timer_gettime` https://linux.die.net/man/2/timer_gettime here since they are not available on Windows. Therefore, we have to stick strictly to the code styles of the posix standard.